lib/Extension/LanguageServer/Server/Method/CompletionMethod.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServer\Server\Method;
+
+use Phpactor\CodeBuilder\Adapter\TolerantParser\TextEdit;
+use Phpactor\Completion\Core\Completor;
+use Phpactor\Completion\Core\Suggestion;
+use Phpactor\Extension\Completion\Application\Complete;
+use Phpactor\Extension\LanguageServer\Response\CompletionItem;
+use Phpactor\Extension\LanguageServer\Response\CompletionList;
+use Phpactor\Extension\LanguageServer\Response\Range;
+use Phpactor\Extension\LanguageServer\Server\Method;
+use Phpactor\MapResolver\Resolver;
+use Phpactor\WorseReflection\Core\Position;
+
+class CompletionMethod implements Method
+{
+    /**
+     * @var Completor
+     */
+    private $completor;
+
+    public function __construct(Completor $completor)
+    {
+        $this->completor = $completor;
+    }
+
+    public function name(): string
+    {
+        return 'textDocument/completion';
+    }
+
+    public function __invoke(array $params): CompletionList
+    {
+        $offset = null;
+        $response = $this->completor->complete(file_get_contents($params['textDocument']['uri']), $offset);
```

Comment:
The LSP expects the server to store the state of the documents - the URI is the identifier. We will need a handler and document store.